counterpartylib/lib/messages/issuance.py

```diff
@@ -124,6 +137,36 @@ def validate (db, source, destination, asset, quantity, divisible, callable_, ca
         if destination:
             problems.append('cannot transfer a non‐existent asset')
 
+    # validate parent ownership for subasset
+    if subasset_longname is not None:
+        cursor = db.cursor()
+        cursor.execute('''SELECT * FROM issuances \
+                          WHERE (status = ? AND asset = ?)
+                          ORDER BY tx_index ASC''', ('valid', subasset_parent))
+        parent_issuances = cursor.fetchall()
+        cursor.close()
+        if parent_issuances:
+            last_parent_issuance = parent_issuances[-1]
+            if last_parent_issuance['issuer'] != source:
+                problems.append('parent asset owned by another address')
+        else:
+            problems.append('parent asset not found')
+
+    # validate subasset issuance is not a duplicate
+    if subasset_longname is not None and not reissuance:
+        cursor = db.cursor()
+        cursor.execute('''SELECT * FROM assets \
+                          WHERE (asset_longname = ?)''', (subasset_longname,))
+        assets = cursor.fetchall()
+        if len(assets) > 0:
```

Comment:
```> 0``` is not necessary, but it's readable. ```if len(assets):``` would return 0/false if length was 0, but IMO good as-is.